addons/mail/static/src/models/message_action_list.js

```diff
@@ -1,187 +1,163 @@
-            this.messageView.replyTo();
+            return clear();
         },
         /**
          * @private
-         * @param {MouseEvent} ev
+         * @returns {FieldCommand}
          */
-        onClickToggleCompact(ev) {
-            this.update({ isCompact: !this.isCompact });
+        _computeEditActionView() {
+            if (this.message && this.message.canBeDeleted) {
+                return insertAndReplace();
+            }
+            return clear();
         },
         /**
          * @private
-         * @param {MouseEvent} ev
+         * @returns {boolean}
          */
-        onClickToggleStar(ev) {
-            this.message.toggleStar();
+        _computeHasReplyToAction() {
+            if (
+                this.messaging && this.messaging.inbox &&
+                this.message && !this.message.isTemporary && !this.message.isTransient &&
+                this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread && (
+                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread === this.messaging.inbox ||
+                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread.model === 'mail.channel'
+                )
+            ) {
+                return true;
+            }
+            return false;
         },
         /**
          * @private
-         * @returns {integer|FieldCommand}
+         * @returns {FieldCommand}
          */
-        _computeActionsCount() {
-            if (this.message){
-                const actions = [
-                    this.hasMarkAsReadIcon,
-                    this.hasReplyIcon,
-                    this.message.canBeDeleted,
-                    this.message.canBeDeleted,
-                    this.message.canStarBeToggled,
-                    this.message.hasReactionIcon,
-                ]
-                return actions.filter(Boolean).length;
+        _computeMarkAsReadActionView() {
+            if (
+                this.messaging && this.messaging.inbox &&
+                this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread &&
+                this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread === this.messaging.inbox
+            ) {
+                return insertAndReplace();
             }
             return clear();
         },
         /**
          * @private
-         * @returns {string}
+         * @returns {FieldCommand}
          */
-        _computeAddReactionText() {
-            return this.env._t("Add a Reaction");
+        _computeReactionActionView() {
+            if (this.message && this.message.hasReactionIcon) {
```

Comment:
on message `actionReaction: one('MessageAction')` ?

then `MessageActionView` identifying fields become `['messageActionListOwner', 'messageAction']`
move the OR identifying to `MessageAction`